File: OP_laba5_part2_c++/src/LowerCaseLetters.cpp
#include "LowerCaseLetters.h"

LowerCaseLetters::LowerCaseLetters(char *value) : ParentString(value) {
    int length = sizeof(value);
    int lower = 0;
    for (int i = 0; i < length; ++i) {
        if(std::islower(value[i])) {
            lower++;
        }
    }
    if (!(length == lower))
    {
        throw std::runtime_error("String should only contain lowercase letters!");
    }
}

void LowerCaseLetters::Sort()
{
    std::sort(value, value+sizeof(value));
    std::reverse(value, value+sizeof(value));
}

File: OP_laba5_part2_sharp/OP_laba5_part2_sharp/LowerCaseLetters.cs
using System;
using System.Linq;
using System.Net.Cache;

namespace OP_laba5_part2_sharp
{
    public class LowerCaseLetters : ParentString
    {
        
        public LowerCaseLetters(params char[] str) : base()
        {
            if (!str.All(Char.IsLower))
            {
                throw new Exception("String should only contain lowercase letters!");
            }
        }

        public override void Sort() 
        {
            base.Sort();
            Array.Reverse(StrValue);
        }
    }
}